scheduler/reconciler/reconcile_node_prop_test.go

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package reconciler
+
+import (
+	"testing"
+	"time"
+
+	"github.com/hashicorp/nomad/nomad/structs"
+	"pgregory.net/rapid"
+)
+
+func TestNodeReconciler_PropTest(t *testing.T) {
+	t.Run("system jobs", rapid.MakeCheck(func(t *rapid.T) {
+		nr := genNodeReconciler(structs.JobTypeSystem, &idGenerator{}).Draw(t, "input")
+		results := Node(nr.job, nr.readyNodes, nr.notReadyNodes,
+			nr.taintedNodes, nr.allocs, nr.terminal, nr.serverSupportsDisconnectedClients)
+		if results == nil {
+			t.Fatal("results should never be nil")
+		}
+		// TODO(tgross): this where the properties under test go
+	}))
+
+	t.Run("sysbatch jobs", rapid.MakeCheck(func(t *rapid.T) {
+		nr := genNodeReconciler(structs.JobTypeSysBatch, &idGenerator{}).Draw(t, "input")
+		results := Node(nr.job, nr.readyNodes, nr.notReadyNodes,
+			nr.taintedNodes, nr.allocs, nr.terminal, nr.serverSupportsDisconnectedClients)
+		if results == nil {
+			t.Fatal("results should never be nil")
+		}
+		// TODO(tgross): this where the properties under test go
+	}))
+
+}
+
+type nodeReconcilerInput struct {
+	job                               *structs.Job
+	readyNodes                        []*structs.Node
+	notReadyNodes                     map[string]struct{}
+	taintedNodes                      map[string]*structs.Node
+	allocs                            []*structs.Allocation
+	terminal                          structs.TerminalByNodeByName
+	serverSupportsDisconnectedClients bool
+}
+
+func genNodeReconciler(jobType string, idg *idGenerator) *rapid.Generator[*nodeReconcilerInput] {
+	return rapid.Custom(func(t *rapid.T) *nodeReconcilerInput {
+		now := time.Now() // note: you can only use offsets from this
+		nodes := rapid.SliceOfN(genNode(idg), 1, 30).Draw(t, "nodes")
```

Comment:
Is there a scenario where we would want to be able to configure/alter this? I am thinking about what happens when there are zero nodes or how this could interact with the `maxAllocs` value.